source/uwp/Renderer/lib/XamlBuilder.cpp

```diff
@@ -1480,9 +1485,15 @@ AdaptiveNamespaceStart
             ComPtr<IShape> ellipseAsShape;
             THROW_IF_FAILED(ellipse.As(&ellipseAsShape));
             // Set Auto, None, and Stretch to Stretch_UniformToFill. An ellipse set to Stretch_Uniform ends up with size 0.
-            if (size == ABI::AdaptiveNamespace::ImageSize::None ||
+            if (isAspectRaitioNeeded)
```

Comment:
Could you add a test file to samples\Tests that demonstrates this and run the test app?